applications/launchpad/gui-react/src/containers/MiningContainer/MiningBox/index.tsx

```diff
@@ -176,19 +179,20 @@ const MiningBox = ({
     },
   }
 
-  const currentState = useMemo(
-    () =>
-      deepmerge.all([
-        defaultConfig,
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        theCurrentStatus ? defaultStates[theCurrentStatus]! : {},
-        theCurrentStatus && statuses && statuses[theCurrentStatus]
-          ? // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            statuses[theCurrentStatus]!
-          : {},
-      ]) as NodeBoxStatusConfig,
-    [theCurrentStatus, nodeState],
-  )
+  const currentState = useMemo(() => {
+    const fromDefaultState = theCurrentStatus
+      ? defaultStates[theCurrentStatus]
+      : {}
+    const fromProps =
+      theCurrentStatus && statuses && statuses[theCurrentStatus]
+        ? statuses[theCurrentStatus]
+        : {}
+    return deepmerge.all([
+      defaultConfig,
+      fromDefaultState || {},
+      fromProps || {},
+    ]) as NodeBoxStatusConfig
+  }, [theCurrentStatus, nodeState, statuses])
```

Comment:
Typescript issue fix. When we enumerate like this: `statuses[theCurrentStatus]`, TS sees this object as possible undefined, even if we wrap this with `if` statement, ie: `if (statuses[theCurrentStatus]) {...}`.

The trick is to assign it to variable (optionally with rescue value):

```
const x = a[b] || {}
```